### PR TITLE
[WIP] Implement manual sorting

### LINKED ISSUE #86: Add ability to sort tasks
At the moment the only way to sort tasks in the UI is to add sequential to-do dates to them, but this kind of abuses the system for non-date tasks.

It'd be good to be able to drag-sort tasks. I gather in the underlying storage backend this would mean reordering the tasks

### PR DESCRIPTION
**Pull request by raimund-schluessler**
This PR aims to implement the possibility to manualls sort the task list by drag-and-drop and will fix #86.

TODOs:
- [x] Enable manually sorting tasks in the list view
- [ ] Fix sorting of tasks when list is ordered in reverse
- [ ] Enable sorting tasks in collection views
- [ ] Enable sorting of subtasks
- [ ] Automatically switch to manual sorting if a task is dragged

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -1562,7 +1562,8 @@ ol[dnd-list] .copy.dndDraggingSource {
 ol[dnd-list] .dndPlaceholder {
   display: block;
   background-color: #ddd;
-  min-height: 37px;
+  min-height: 36px;
+  height: 36px;
```
Comment (ccoenen) [inline css/style.css]: this change does not seem related to the PR.

File: js/public/app.js [hunk 2]
```diff
@@ -4262,6 +4291,34 @@ angular.module('Tasks').factory('VTodo', ['$filter', 'ICalFactory', 'RandomStrin
...
+						second: 0,
+						isDate: false
+					})
+				).toSeconds();
+			}
+			return +order;
+		},
+		set manualSort(order) {
+			var vtodos = this.components.getAllSubcomponents('vtodo');
+			vtodos[0].updatePropertyWithValue('x-apple-sort-order', order);
```
Comment (ccoenen) [inline js/public/app.js]: I am not sure if `X-APPLE...` is a property I'd like in my VTODOs.
Maybe there's something in https://tools.ietf.org/html/rfc3648 that would also do the trick? (comment applies to https://github.com/owncloud/tasks/pull/358/files#diff-37bb856dc04e8c65595d2746df449090R4296 as well, of course)

Comment (raimund-schluessler) [inline js/public/app.js]: This was discussed in https://github.com/owncloud/tasks/issues/86 already. There is no standard key for sorting VTODOs. So we would have to use our own key anyway. And the benefit of ```X-APPLE-SORT-ORDER``` is that it syncs to iOS devices. It also does not break anything else.

**Comment by raimund-schluessler** [comment]
See https://github.com/nextcloud/tasks/pull/4
